features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/components/MaxBackendTps.jsx

```diff
@@ -0,0 +1,186 @@
+                            >
+                                <HelpOutline className={classes.iconSpace} />
+                            </Tooltip>
+                        </Typography>
+                    </ExpansionPanelSummary>
+                    <ExpansionPanelDetails className={classes.expansionPanelDetails}>
+                        <FormControl component='fieldset'>
+                            <FormLabel component='legend'>Maximum Throughput</FormLabel>
+                            <RadioGroup
+                                aria-label='change-max-TPS'
+                                name={api.maxTps}
+                                value={api.maxTps === null}
+                                onChange={(event) => {
+                                    configDispatcher({
+                                        action: 'maxTps',
+                                        value:
+                                            event.target.value === 'false' ? { production: null, sandbox: null } : null,
+                                    });
+                                }}
+                                row
+                            >
+                                <FormControlLabel
+                                    value
+                                    control={<Radio color='default' />}
+                                    label='Unlimited'
+                                    labelPlacement='end'
+                                />
+                                <FormControlLabel
+                                    value={false}
+                                    control={<Radio color='default' />}
+                                    label='Specify'
+                                    labelPlacement='end'
+                                />
+                            </RadioGroup>
+                        </FormControl>
+                        <Collapse in={api.maxTps !== null}>
+                            <Grid item xs={12} style={{ marginBottom: 15, position: 'relative' }}>
+                                <TextField
+                                    id='outlined-name'
+                                    label='Max Production TPS'
+                                    margin='normal'
+                                    variant='outlined'
+                                    onChange={(event) => {
+                                        configDispatcher({
+                                            action: 'maxTps',
+                                            value: {
+                                                production: event.target.value,
+                                                sandbox: api.maxTps !== null ? api.maxTps.sandbox : null,
+                                            },
+                                        });
+                                    }}
+                                    value={api.maxTps !== null ? api.maxTps.production : ''}
```

Comment:
I don't think we need to have a null check here, Ideally, this component should not render if the `maxTps` value is empty.